docs/questions/query-builder/expressions/now.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Now
+---
+
+# Now
+
+`now` returns the current date and time using your Metabase [report timezone](../configuring-metabase/settings#report-timezone).
+
+| Syntax         | Example                     |
+|----------------|-----------------------------|
+|`now`           | November 22, 2022, 16:00:00 |
+
+## Creating conditional logic using the current date or time
+
+Let's say you have some project data, and you want to add a status column for each task. Assume today's date and time is November 22, 2022, 16:00:00.
+
+| Task     | Start                       | Deadline                    | Status          |
+|----------|-----------------------------|-----------------------------|-----------------|
+| Draft    | November 1, 2022, 12:00:00  | November 30, 2022, 16:00:00 | In progress     |
+| Review   | November 15, 2022, 16:00:00 | November 19, 2022, 16:00:00 | Needs extension |
+| Edit     | November 22, 2022, 12:00:00 | November 22, 2022, 16:00:00 | DUE RIGHT NOW!  |
+
+To mark a task in progress, you'd use the expression:
+
+```
+now >= [Start] AND now < [Deadline] 
+```
+
+To check if you need to ask for an extension:
+
+```
+now >= [Start] AND now >= [Deadline]
+```
+
+If you're looking for an adrenaline rush (and you have real-time data), you can flag the tasks that are due _right this second_:
+
+```
+now = [Deadline]
+```
+
+To set up the **Status** column that combines all three situations above, you'd wrap everything in a `case` expression:
+
+```
+case(now >= [Start] AND now < [Deadline], "In progress",
+     now >= [Start] AND now >= [Deadline], "Needs extension",
+     now = [Deadline], "DUE RIGHT NOW!")
+```
+
+## Accepted data types
+
+| [Data type](https://www.metabase.com/learn/databases/data-types-overview#examples-of-data-types) | Works with `now`  |
+| ----------------------- | -------------------- |
+| String                  | ❌                   |
+| Number                  | ❌                   |
+| Timestamp               | ✅                   |
+| Boolean                 | ❌                   |
+| JSON                    | ❌                   |
+
+This table uses `timestamp` and `datetime` interchangeably---just make sure that your dates and times aren't stored as strings or numbers in your database.
+
+## Limitations
+
+`now` might not actually be _now_ if you don't live in the same timezone as your Metabase report timezone.
```

Comment:
Metabase, or the queried database? What about when querying a model that used `convertTimezone` to prep the data?

Reply:
@calherries if you could confirm the note here and my understanding below, that'd be swell :)

The spec says that `now` should return the equivalent of `convertTimezone(what_the_db_returns, reporting_timezone)`.

If your Metabase reporting timezone is EST, `now` translates to:

```sql
SELECT CURRENT TIMESTAMP AT TIME ZONE 'EST` AS now
```

Let's say your database has a column called `datetime_utc`. You can make a model with a column `datetime_pst` by writing `convertTimezone(datetime_utc, 'PST')`, which should run as:

```sql
SELECT datetime_utc AT TIME ZONE 'PST' AS datetime_pst
```

Then, if you write an expression using the model to check if `now > datetime_pst`:

```sql
SELECT CURRENT_TIMESTAMP AT TIME ZONE 'EST' > datetime_utc AT TIME ZONE 'PST'
```

The logic will check if the current date and time in EST is greater than `datetime_utc` mapped to PST.

I'll add a note that you should name custom datetime columns with the time zone if possible 😂

Reply:
I'm not 100% sure what you're asking here, since now translates to something slightly different for every driver. I assume this is postgres though.

> now should return the equivalent of convertTimezone(what_the_db_returns, reporting_timezone)

This isn't really true, and it's not the best way to think about it how it works. "what the db returns" ,depending on the driver, can already be in the reporting_timezone. If it's not, we convert it for the user.
It's best to think that `now` returns the current time in the reporting timezone. It's got nothing to do with `convertTimezone`.

Unfortunately, it's not quite as simple as that, because there are slight differences when the data is at rest, for example when the question is turned into a model.

If a database supports a timestamp type with time zones at rest, the result of `now()` will be a "datetime with time zone" type. Presto fits into this category.

If a database supports a timestamp type with offsets at rest, the result of `now()` will be a "zoned datetime" type. It will have an offset from UTC, not a time zone. Snowflake fits into this category.

If a database supports neither, then the result of `now()` will be a "datetime without timezone" type, and it will be stored in UTC.

But none of this matters to the user of the GUI editor. Only the SQL user might notice if they inspect the types.

The question you raise is more related to convertTimezone in general, rather than now(). And this is where users could get very confused. It's a deeply complicated topic.

The key problem with convertTimezone is that the result is converted to a type without a time zone attached. And we usually assume that datetimes without time zones are by default in the reporting_timezone, except sometimes in native SQL. There are probably some thorny edge cases and bugs we haven't considered with this.

Your example is tricky because it has a column with `datetime_utc` and a reporting timezone that is not UTC.

If it's postgres, I'm assuming `datetime_utc` is a `timestamp without time zone` column. (If it were a `timestamp with time zone` column it couldn't be in UTC, because the time zone at rest is the database timezone). So in order to create a datetime_pst column, yes, the user should do `convertTimezone(datetime_utc, 'PST', 'UTC')`. The 'UTC' argument needs to be given because the `datetime_utc` column doesn't have a time zone. So leaving it out would make us assume the column's time zone is implicitly the reporting time zone. Then, the result of convertTimezone would be another column of type `timestamp without time zone`.

If you did a comparison like `now < datetime_pst` where datetime_pst is a model column, that would translate into:
```
SELECT NOW() > datetime_pst;
```

The problem is `datetime_pst` is a timestamp without time zone, and in comparisons with `>` timestamp without time zones are assumed to be in the database time zone. If we were writing native SQL, we would want to write 
```
SELECT NOW() AT TIME ZONE 'PST'  > datetime_pst;
```
One way to write this with a custom expression is `convertTimezone(now, "PST") < datetime_pst`.

An alternative is to convert `datetime_pst` into the reporting_timezone before comparison:
```
now < convertTimezone(datetime_pst, 'EST', 'PST')
```

So, trying to summarize:
- `now`'s result implicitly takes the time zone of the reporting timezone if set
- `convertTimezone`'s result is a type without a time zone
- If you want to do comparisons between a timestamp with time zone and timestamp without time zone, you should convert them to the same timezone first.

So that's some of what users need to know. I'm sorry I didn't have enough time to write a shorter explanation.

Reply:
cc @qnkhuat If you have anything to add on convertTimezone

Reply:
@calherries This is incredibly helpful, thank you for writing it up 🙏